packages/patternfly-4/react-charts/src/components/ChartDonut/ChartDonut.js

```diff
@@ -1,21 +1,164 @@
-  '': PropTypes.any
+  '': PropTypes.any,
+  /**
+   * The height props specifies the height the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  height: PropTypes.number,
+  /**
+   * The subtitle for the donut chart
+   */
+  subTitle: PropTypes.string,
+  /*
+   * Specifies the theme color; blue (default), green, or multi-color. Overridden by the theme property.
+   */
+  themeColor: PropTypes.string,
+  /*
+   * Specifies the theme variant; 'dark' or 'light' (default). Overridden by the theme property.
+   */
+  themeVariant: PropTypes.string,
+  /**
+   * The title for the donut chart
+   */
+  title: PropTypes.string,
+  /**
+   * The width props specifies the width of the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number
 };
 
+class ChartDonut extends React.Component {
+  titleRef = React.createRef();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      height: this.props.height || 0,
+      width: this.props.width || 0
+    };
+  }
+
+  componentDidMount() {
+    // Skip resize handler if height and width were provided
+    if (!(this.state.height || this.state.width)) {
+      setTimeout(() => {
+        window.addEventListener('resize', this.handleResize);
+      });
+      this.handleResize();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  getSubTitle = () => {
+    const { subTitle } = this.props;
+    const { height, width } = this.state;
+
+    if (subTitle && height && width) {
+      const xVal = width / 2 - subTitle.length / 2;
+      const yVal = height / 2 + 10;
+
+      return (
+        <ChartLabel
+          style={styles.label.subTitle}
+          text={subTitle}
+          textAnchor="middle"
+          verticalAnchor="middle"
+          x={xVal}
+          y={yVal}
+        />
+      );
+    }
+    return null;
+  };
+
+  getTitle = () => {
+    const { title } = this.props;
+    const { height, width } = this.state;
+
+    if (title && height && width) {
+      const xVal = width / 2 - title.length / 2;
+      const yVal = height / 2 - 10;
```

Comment:
It there is no subtitle (perhaps when a donut is just showing a percentage), then the title will be off center when subtracting 10px from the yVal.

Reply:
true